### PR TITLE
Allow adding custom classes

### PR DESCRIPTION
**Pull request by allison-strandberg**
Add the `data-chardinjs-class` and `data-chardinjs-helper-class` attributes, for adding custom classes to the element and tooltip when the overlay is displayed. Consistently namespace data attributes so they all include the string "chardin". Perform some general maintenance, such as resolving rubocop issues and adding Gemfile.lock and sourcemap files.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,23 +28,25 @@ There's a [`chardinjs-rails`](https://github.com/heelhook/chardin.js-rails) gem.
...
 
 ```HTML
-<img src="img/chardin.png" data-intro="An awesome 18th-century painter, who found beauty in everyday, common things." data-position="right" />
+<img src="img/chardin.png" data-chardinjs-intro="An awesome 18th-century painter, who found beauty in everyday, common things." data-chardinjs-position="right" data-chardinjs-class="white-border" data-chardinjs-helper-class="italic" />
 ```
 
 You can also run Chardin in sequenced mode, where one element will be displayed at a time, moving on to the next with a mouse click (or automatically after a set delay).
-Add `data-chardin-sequenced="true"` entry to the body tag. Also add `data-chardin-auto="true"` and `data-chardin-delay="100"` for automatic movement through the elements. Delay is in milliseconds.
-The default sequence is as loaded by the DOM, but this can be overridden using the tag `data-sequence` with a number. 
+Add `data-chardinjs-sequenced="true"` entry to the body tag. Also add `data-chardinjs-auto="true"` and `data-chardinjs-delay="100"` for automatic movement through the elements. Delay is in milliseconds.
```
Comment (waissbluth) [inline README.md]: how about keeping "chardin" instead of "chardinjs" for some extra backward compatibility? here and everywhere

**Review by waissbluth — COMMENTED** [review]
while you were here did you get a chance to assess how dependent on jquery the library is?

File: README.md (hunk 1 above)
Comment (allison-strandberg) [reply README.md]: will use "chardin" for data attributes and keep "chardinjs" for css classes.

File: chardinjs.js [hunk 2]
```diff
@@ -7,16 +7,16 @@
         var chardinJs;
         chardinJs = (function () {
             function chardinJs(el) {
-                this.data_attribute = 'data-intro';
+                this.data_attribute = 'data-chardin-intro';
                 this.chardinCssClasses = ["chardinjs-helper-layer", "chardinjs-show-element", "chardinjs-relative-position"];
                 this.$el = $(el);
-                this.sequenced = this.$el.data('chardin-sequenced') ? true : false;
+                this.sequenced = this.$el.data('chardinjs-sequenced') ? true : false;
```
Comment (waissbluth) [inline chardinjs.js]: chardin-sequenced?

File: chardinjs.js [hunk 3]
```diff
@@ -139,7 +147,7 @@
 
                 this.$el.get()[0].appendChild(overlay_layer);
 
-                this.$el.find("#chardin-mask").fadeIn();
+                this.$el.find("#chardinjs-mask").fadeIn();
```
Comment (waissbluth) [inline chardinjs.js:150]: #chardin-mask?

**Review by waissbluth — CHANGES_REQUESTED** [review]

File: chardinjs.js (hunk 3 above)
Comment (allison-strandberg) [reply chardinjs.js:150]: So element IDs should follow the data attribute convention "chardin" and not the css class convention "chardinjs"?

**Pull request merged**